### PR TITLE
SDIT-1349 Add Activities prison pay band check to UI

### PR DESCRIPTION
**Pull request by mikehalmamoj**
New feature is **DPS has no pay bands for prison WWI.**.  Also tidied up the **Activate prison** button.
image

### CODE REVIEW

File: server/routes/activitiesMigration/activitiesMigrationController.ts [hunk 1]
```diff
@@ -112,7 +112,15 @@ export default class ActivitiesMigrationController {
...
       }),
-    ]).then(([estimatedCount, dlqCount, incentiveLevels, nomisFeatureSwitchOn, dpsPrisonRollout]) => {
+
+      this.activitiesService.checkPrisonPayBandsExist(prisonId, context(res)).catch(error => {
+        errors.push({
+          text: `Failed to check if prison ${prisonId} has pay bands in DPS due to error: ${error.message}`,
+          href: '',
+        })
+        return null
+      }),
```
Comment (mikehalmamoj) [inline server/routes/activitiesMigration/activitiesMigrationController.ts:122]: The main change is to check if there are any pay bands set up for the prison in DPS.

File: server/views/pages/activities/startActivitiesMigrationPreview.njk [hunk 2]
```diff
@@ -42,21 +42,21 @@
...
                             {% if not form.prisonSwitchedOnDps %}
                                 <p class="govuk-notification-banner__heading" id="dpsFeatureSwitch">
                                     DPS feature switch not turned on for {{ form.prisonId }}.
                                 </p>
                             {% endif %}
+                                {% if not form.dpsPayBandsExist %}
+                                <p class="govuk-notification-banner__heading" id="dpsPayBands">
+                                    DPS has no pay bands for prison {{ form.prisonId }}.
+                                </p>
+                            {% endif %}
```
Comment (mikehalmamoj) [inline server/views/pages/activities/startActivitiesMigrationPreview.njk:59]: And show a warning if there are no pay bands.

File: integration_tests/e2e/startActivitiesMigration.cy.ts [hunk 3]
```diff
@@ -131,7 +132,8 @@ context('Start Activities Migration', () => {
       previewPage.incentiveLevels().contains('BAS,STD,ENH')
       previewPage.nomisFeatureSwitch().should('not.exist')
       previewPage.activateFeatureSwitch().should('not.exist')
-      previewPage.dpsFeatureSwitch().should('not.exist')
```
Comment (mikehalmamoj) [inline integration_tests/e2e/startActivitiesMigration.cy.ts:134]: Fixed this assertion by configuring the activities URL correctly.

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**